src/play.jl

```diff
@@ -54,57 +136,139 @@ function bincounts!(
     return counts
 end
 
-nextguess(gp::GamePool) = gp.pool[gp.guess[]]
+function bincounts!(gp::GamePool, k)
+    bincounts!(gp.counts, gp.active, view(gp.allscores, :, k))
+    return gp
+end
 
 """
-    reset!(gp::GamePool)
+    entropybase2(probs::AbstractVector{<:AbstractFloat})
+    entropybase2(gp::GamePool)
+
+Return the base-2 entropy of `probs` or `gp.probs`.  This is the entropy measured in bits.
 
-Return `gp` with its `active`, `guess`, and `expectedpoolsize` fields reset to initial values.
+See the [Wikipedia entry](https://en.wikipedia.org/wiki/Entropy_(information_theory))
+for the definition of entropy in information theory.
+
+`probs` is assumed to be a discrete probability distribution.  That is the conditions
+`all(0 .≤ probs .≤ 1) && sum(probs) ≈ 1` are assumed (but not checked).
 """
-function reset!(gp::GamePool)
-    fill!(gp.active, true)
-    gp.guess[] = gp.firstguess
-    gp.expectedpoolsize[] = gp.firstexpectedpoolsize
-    return gp
+function entropybase2(probs::AbstractVector{<:AbstractFloat})
+    return -sum(x -> iszero(x) ? zero(x) : x * log2(x), probs)
```

Comment:
Using StatsFuns.jl, you could write this as
```suggestion
    return -sum(x -> xlogx(x) / logtwo, probs)
```
though I don't know if the extra division will be faster or slower than the branch, esp since there's probably a branch in `xlogx`

Reply:
There is a branch in `xlogx` but that's not important.  Your code is cleaner.